source/extensions/common/dynamic_forward_proxy/dns_cache_impl.cc

```diff
@@ -23,17 +24,18 @@ DnsCacheImpl::DnsCacheImpl(
       resolver_(main_thread_dispatcher.createDnsResolver({}, false)), tls_slot_(tls.allocateSlot()),
       scope_(root_scope.createScope(fmt::format("dns_cache.{}.", config.name()))),
       stats_(generateDnsCacheStats(*scope_)),
+      resource_manager_(
+          *scope_, loader, config.name(),
+          config.has_dns_cache_circuit_breaker()
+              ? config.dns_cache_circuit_breaker()
+              : envoy::extensions::common::dynamic_forward_proxy::v3::DnsCacheCircuitBreakers()),
```

Comment:
redundant, `config.dns_cache_circuit_breaker()` will return a default instance if it isn't set.